File: solution/backtracking/15651/main.cpp
// Authored by : tony9402
// Co-authored by : -
// Link : http://boj.kr/0a5ce31db1ff47cf8190b0275e74bef8
#include<bits/stdc++.h>

using namespace std;

int choose[10], N, M;

void dfs(int cnt) {
    if(cnt == M) {
        for(int i=0;i<M;i++) {
            cout << choose[i] << ' ';
        }
        cout << '\n';
        return;
    }

    for(int i=1;i<=N;i++) {
        choose[cnt] = i;
        dfs(cnt + 1);
    }
}

int main(){
    ios::sync_with_stdio(false);
    cin.tie(0);

    cin >> N >> M;
    dfs(0);

    return 0;
}

File: solution/backtracking/15651/main.py
# Authored by : tony9402
# Co-authored by : -
# Link : http://boj.kr/4631083fc0c04817af9b1020e2396ddc
import sys

def input():
    return sys.stdin.readline().rstrip()

N, M = map(int, input().split())
choose = [ 0 for _ in range(10) ]

def dfs(idx, cnt):
    global N, M
    if cnt == M:
        for idx in range(cnt):
            print(choose[idx], end=' ')
        print()
        return

    for i in range(1, N + 1):
        choose[cnt] = i
        dfs(i + 1, cnt + 1)

dfs(1, 0)

